packages/flutter_tools/lib/src/cache.dart

```diff
@@ -706,22 +710,26 @@ class FlutterWebSdk extends CachedArtifact {
   @override
   Future<void> updateInner(ArtifactUpdater artifactUpdater) async {
     String platformName = 'flutter-web-sdk-';
-    if (globals.platform.isMacOS) {
+    if (_platform.isMacOS) {
       platformName += 'darwin-x64';
-    } else if (globals.platform.isLinux) {
+    } else if (_platform.isLinux) {
       platformName += 'linux-x64';
-    } else if (globals.platform.isWindows) {
+    } else if (_platform.isWindows) {
       platformName += 'windows-x64';
     }
     final Uri url = Uri.parse('${cache.storageBaseUrl}/flutter_infra/flutter/$version/$platformName.zip');
+    if (location.existsSync()) {
+      location.deleteSync(recursive: true);
```

Comment:
Should this ignore exceptions?

Reply:
I think if we fail to delete this, we'd also likely fail to copy files over it